ios/Sources/TimetableFeature/TimetableFeature.swift

```diff
@@ -23,25 +36,42 @@ public struct TimetableState: Equatable {
 }
 
 public enum TimetableAction {
+    case refresh
+    case refreshResponse(Result<[AnyTimetableItem], KotlinError>)
     case selectedPicker(SelectedType)
     case content(TimetableContentAction)
     case hideDetail
 }
 
 public struct TimetableEnvironment {
-    public init() {}
+    public let timetableRepository: TimetableRepositoryProtocol
+
+    public init(
+        timetableRepository: TimetableRepositoryProtocol
+    ) {
+        self.timetableRepository = timetableRepository
+    }
 }
 
-public let timetableReducer = Reducer<TimetableState, TimetableAction, TimetableEnvironment> { state, action, _ in
+public let timetableReducer = Reducer<TimetableState, TimetableAction, TimetableEnvironment> { state, action, environment in
     switch action {
+    case .refresh:
+        return environment.timetableRepository.timetableContents()
+            .receive(on: DispatchQueue.main)
+            .catchToEffect()
+            .map(TimetableAction.refreshResponse)
+    case let .refreshResponse(.success(items)):
+        state.timetableItems = items.sorted { $0.startsAt < $1.startsAt }
+        return .none
+    case let .refreshResponse(.failure(error)):
+        print(error.localizedDescription)
```

Comment:
why not show user that error occurred.
this might help you for simple way https://github.com/DroidKaigi/conference-app-2021/blob/74dfe5dc1c70d06631b3128b1b21b8d06612c249/ios/Sources/AppFeature/AppScreen.swift#L64